lib/rules/curly.js

```diff
@@ -97,10 +97,11 @@ module.exports = {
          * @private
          */
         function isOneLiner(node) {
-            const first = sourceCode.getFirstToken(node),
-                last = sourceCode.getLastToken(node);
+            const first = sourceCode.getFirstToken(node);
+            const last = sourceCode.getLastToken(node);
+            const lastExcludingSemicolon = astUtils.isSemicolonToken(last) ? sourceCode.getTokenBefore(last) : last;
 
-            return first.loc.start.line === last.loc.end.line;
+            return first.loc.start.line === lastExcludingSemicolon.loc.end.line;
```

Comment:
What will happen with an empty statement (only the `;`).  For example, would this code have a warning:

```js
/* eslint curly: ["error", "multi-or-nest"] */

if (foo) 
;
```

Reply:
Yes, that code would have produced a warning. I just added
```js
if (node.type === "EmptyStatement") {
    return true;
}
```
at the beginning of the function to fix it.

Reply:
Thanks!